src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java

```diff
@@ -0,0 +1,272 @@
+        requestSpec.header("Content-Type", "application/json;charset=UTF-8");
+        return requestSpec;
+    }
+
+    protected static String setBodyLoanProduct() {
+        // Generating product name and shortname
+        String name = new StringBuilder().append("loan").append(getAlphaNumericString(4)).toString();
+        String shortName = getAlphaNumericString(4);
+
+        AllowAttributeOverrides allowAttributeOverrides = new AllowAttributeOverrides(true, true, "interest-principal-penalties-fees-order-strategy", true, true, true, true, true);
+
+        LoanProduct loanProduct = new LoanProduct("USD", "false", false, "2", "2", 2, 3, 1, 0, 6, new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), false, "1", 1, 1, true, 0, 2, 1, false, false, name, shortName, "12", "100", "3000", "21000", "3", "36", "60", "5.9", 19, "35.9", "1", true, true, "7", "90", 1, new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), "interest-principal-penalties-fees-order-strategy", allowAttributeOverrides, "en", "dd MMMM yyyy", new ArrayList<>(), false, null);
+
+        return loanProduct.toString();
+    }
+
+
+    protected static String setBodyLoanAccount() throws JsonProcessingException {
+
+        String date = getCurrentDate();
+        setcurrentDate(date);
+
+        CreatePayerClientResponse createPayerClientResponse = objectMapper.readValue(
+                responsePayerClient, CreatePayerClientResponse.class);
+        LoanProductResponse loanProductResponse = objectMapper.readValue(
+                responseLoanProduct, LoanProductResponse.class);
+
+        LoanAccountData loanAccountData = new LoanAccountData(createPayerClientResponse.getClientId(), Integer.parseInt(loanProductResponse.getResourceId()), new ArrayList<>(), 7800, 12, 2, 12, 1, 2, 8.9, 1, false, 0, 0, false, 7, "interest-principal-penalties-fees-order-strategy", new ArrayList<>(), currentDate, "en", "dd MMMM yyyy", "individual", currentDate, currentDate);
+
+        return loanAccountData.toString();
+    }
+
+    private static String getCurrentDate() {
+        Date date = new Date();
+        return new SimpleDateFormat("dd MMMM yyyy").format(date);
+    }
+
+    protected static String setBodyLoanApprove(int amount) {
+        LoanApprove loanApprove = new LoanApprove(currentDate, amount, currentDate, new ArrayList<>(), "en", "dd MMMM yyyy");
+        return loanApprove.toString();
+    }
+
+    protected static Object setBodyLoanRepayment(String amount) {
+        LoanRepaymentDTO loanRepaymentDTO = new LoanRepaymentDTO(currentDate, "1", amount, "en", "dd MMMM yyyy");
+        return loanRepaymentDTO.toString();
+    }
+
+    protected static RequestSpecification setHeadersSavings(RequestSpecification requestSpec) {
+        requestSpec.header("fineract-platform-tenantid", tenant);
+        requestSpec.header("accept", "application/json, text/plain, */*");
```

Comment:
use minimal and configurable headers. also reuse the methof

Reply:
they have different letter case for headers that's why can'y reuse for loan and savings.

Reply:
Http is not case sensitive, so try reusing